backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -1293,11 +1333,40 @@ public void updateV2VJobInfos(List<V2VJobInfo> v2vJobInfos) {
         }
     }
 
+
     private void initAvailableCpus() {
         List<VdsCpuUnit> cpuTopology = cachedVds.getCpuTopology();
-        if (cpuTopology == null) {
+        if (cpuTopology == null || cachedVds.getVdsmCpusAffinity() == null) {
             return;
         }
         this.cpuTopology = cpuTopology;
+        this.cpuTopology.stream().filter(cpu -> cpu.getCpu() == Integer.parseInt(cachedVds.getVdsmCpusAffinity())).forEach(cpu -> cpu.setVmId(Guid.SYSTEM));
+    }
+
+    public List<VdsCpuUnit> getCpuTopology() {
+        List<VdsCpuUnit> clone = new ArrayList<>();
+        cpuTopology.forEach(cpu -> clone.add(cpu.clone()));
+        return clone;
+    }
+
+    public void unpinVmCpus(VM vm) {
+        String pinning = vm.getCpuPinningPolicy() == CpuPinningPolicy.MANUAL ? vm.getCpuPinning() : vm.getCurrentCpuPinning();
+        if (pinning != null && !pinning.isBlank()) {
+            synchronized (this) {
+                cpuTopology.stream().filter(cpu -> CpuPinningHelper.getAllPinnedPCpus(pinning).contains(cpu.getCpu())).forEach(cpu -> {
```

Comment:
we'll call CpuPinningHelper.getAllPinnedPCpus for each cpu on the host.. how about calling it once?